code/modules/research/nanites/nanite_programs/buffing.dm

```diff
@@ -137,3 +137,36 @@
 	to_chat(host_mob, "<span class='notice'>Your body feels lighter and your legs feel relaxed!</span>")
 	host_mob.set_resting(FALSE)
 	host_mob.reagents.add_reagent(/datum/reagent/medicine/amphetamine, 3)
+
+/datum/nanite_program/armblade
+	name = "Nanite Blade"
+	desc = "The nanites form a sharp blade around the user's arm when activated."
+	use_rate = 1
+	rogue_types = list(/datum/nanite_program/necrotic, /datum/nanite_program/skin_decay)
+	var/obj/item/melee/arm_blade/nanite/blade
+
+/datum/nanite_program/armblade/enable_passive_effect()
+	. = ..()
+	if(blade)
+		QDEL_NULL(blade)
+	if(!host_mob)
+		return
+	blade = new(host_mob)
+	host_mob.dropItemToGround(host_mob.get_active_held_item())
+	if(!host_mob.put_in_hands(blade))
+		to_chat(host_mob, "<span class='danger'>You feel an intense pain as your nanites fail to form a blade!</span>")
+		host_mob.adjustBruteLoss(10)
```

Comment:
Could you in theory, have 2 arm blade programs and then just turn on the first blade and spam the second to kill someone in record time?

Reply:
You can only have one of each nanite program type by default (unless otherwise specified, like with sensors, dermal button, or host scan)